test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -25,994 +43,1415 @@ const noop = () => {};
 // .. code:: javascript
 
 //   Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk
-
-describe('Client Side Encryption Prose Tests', function () {
-  const metadata = { requires: { clientSideEncryption: true, mongodb: '>=4.2.0' } };
-  const dataDbName = 'db';
-  const dataCollName = 'coll';
-  const dataNamespace = `${dataDbName}.${dataCollName}`;
-  const keyVaultDbName = 'keyvault';
-  const keyVaultCollName = 'datakeys';
-  const keyVaultNamespace = `${keyVaultDbName}.${keyVaultCollName}`;
-
-  const shared = require('../shared');
-  const dropCollection = shared.dropCollection;
-  const APMEventCollector = shared.APMEventCollector;
-
-  const LOCAL_KEY = Buffer.from(
-    'Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk',
-    'base64'
-  );
-
-  describe('Data key and double encryption', function () {
-    // Data key and double encryption
-    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    // First, perform the setup.
-    beforeEach(function () {
-      const mongodbClientEncryption = this.configuration.mongodbClientEncryption;
-
-      // #. Create a MongoClient without encryption enabled (referred to as ``client``). Enable command monitoring to listen for command_started events.
-      this.client = this.configuration.newClient({}, { monitorCommands: true });
-
-      this.commandStartedEvents = new APMEventCollector(this.client, 'commandStarted', {
-        exclude: [LEGACY_HELLO_COMMAND]
-      });
-
-      const schemaMap = {
-        [dataNamespace]: {
-          bsonType: 'object',
-          properties: {
-            encrypted_placeholder: {
-              encrypt: {
-                keyId: '/placeholder',
-                bsonType: 'string',
-                algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random'
-              }
-            }
+// TODO: NODE-3927 - these cannot run in lb mode but are not skipping based on metadata.
+if (!process.env.LOAD_BALANCER) {
```

Comment:
Interesting that the metadata is not working, could we try skipping inside a `beforeEach` hook if possible? Currently we have a consistent test count across all runs, it would be nice to keep that for LB too.

It appears we currently don't run FLE against lb because we don't have the `CSFLE_KMS_PROVIDERS` var defined in that environment, I don't see anything in the evergreen changes that would've brought it in, what made this crop up?

Reply:
I've updated to skip in the before/beforeEach blocks now and removed this conditional.

The `CSFLE_KMS_PROVIDERS` variable is not in the env, but the metadata isn't skipping for some reason and the test attempts to JSON.parse that env variable anyways causing a parse error on the string 'NOT_PROVIDED'. So I tried to change that string to be empty just to get around it but the before hooks are still running. I suspect there's some funkiness with mocha and async functions but I haven't been able to confirm what the exact issue yet is, and didn't want to rewrite all the tests without async await since it made it way easier to read. But I can do that if we want.